crates/iceberg/src/spec/schema.rs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! This module defines schema in iceberg.
+
+use crate::spec::datatypes::{StructField, StructType};
+
+const DEFAULT_SCHEMA_ID: i32 = 0;
+
+/// Defines schema in iceberg.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Schema {
+    r#struct: StructType,
+    schema_id: i32,
+    highest_field_id: i32,
+}
```

Comment:
I think the `identifier-field-ids` are missing: https://iceberg.apache.org/spec/#identifier-field-ids